tests/models/test_cpu.py

```diff
@@ -190,14 +192,15 @@ def test_default_logger_callbacks_cpu_model(tmpdir):
 def test_running_test_after_fitting(tmpdir):
     """Verify test() on fitted model."""
     class ModelTrainValTest(BoringModel):
+        def validation_step(self, *args, **kwargs):
+            output = super().validation_step(*args, **kwargs)
+            self.log('val_loss', output['x'])
+            return output
 
-        def validation_epoch_end(self, outputs) -> None:
-            val_loss = torch.stack([x["x"] for x in outputs]).mean()
-            self.log('val_loss', val_loss)
-
-        def test_epoch_end(self, outputs) -> None:
-            test_loss = torch.stack([x["y"] for x in outputs]).mean()
-            self.log('test_loss', test_loss)
+        def test_step(self, *args, **kwargs):
+            output = super().test_step(*args, **kwargs)
+            self.log('test_loss', output['y'])
+            return output
```

Comment:
noob question: why do we `self.log` in this subclass here instead of in the superclass `BoringModel`? `self.log` seems quite general to me. Is it for efficiency?

Reply:
yeah maybe since log is not required everywhere and adding that in BoringModel might create an overhead.

Reply:
That makes sense. Thanks :)